components/create_swagger_config.py

```diff
@@ -600,14 +616,35 @@ def CreateSwaggerOperation(
     # create responses
     swaggerResponses = {}
     if isiRespSchema is not None:
-        objRef = IsiSchemaToSwaggerObjectDefs(
-                    isiRespObjNameSpace, isiRespObjName, isiRespSchema,
-                    objDefs, classExtPostFix,
-                    isResponseObject=True)
+        try:
+            responseType = isiRespSchema["type"]
+        except KeyError:
+            # there is some code in IsiSchemaToSwaggerObjectDefs that handles
+            # response schema's that don't have a "type" so just force type to
+            # be an "object" so that IsiSchemaToSwaggerObjectDefs can fix.
+            responseType = "object"
         # create 200 response
         swagger200Resp = {}
-        swagger200Resp["description"] = isiInputArgs["description"]
-        swagger200Resp["schema"] = { "$ref": objRef }
+        # the "type" of /4/protocols/smb/shares and /3/antivirus/servers is a
+        # list of objects, so treat them the same as "type" == "object"
+        if responseType == "object" or type(responseType) == list:
+            objRef = \
+                    IsiSchemaToSwaggerObjectDefs(
+                            isiRespObjNameSpace, isiRespObjName, isiRespSchema,
+                            objDefs, classExtPostFix,
+                            isResponseObject=True)
+            swagger200Resp["description"] = isiInputArgs["description"]
+            swagger200Resp["schema"] = { "$ref": objRef }
+        else:
+            # the "type" of /2/cluster/external-ips is array
```

Comment:
Here's some new code that I added to make the /2/cluster/external-ips end point work. It was broken because it uses a list instead of a object.